.circleci/config.yml

```diff
@@ -0,0 +1,86 @@
+---
+version: 2
+jobs:
+  unit-tests:
+    docker: &docker
+    - image: python:3.8
+    steps:
+    - checkout
+    - &restore_cache
+      restore_cache:
+        keys:
+        # when lock files change, use increasingly general
+        # patterns to restore cache
+        - &cache_key
+          # yamllint disable-line rule:line-length
+          python-3.8-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-{{ checksum "requirements.txt" }}
+          # yamllint disable-line rule:line-length
+        - python-3.8-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-
+        - python-3.8-packages-v1-{{ .Branch }}-
+        - python-3.8-packages-v1-master-
+        - python-3.8-packages-v1-
+    - &build
+      run:
+        name: Build
+        command: |
+          python3.8 -m venv venv/
+          venv/bin/pip install pip-tools --constraint requirements.in
+          venv/bin/pip-sync
+    - run:
+        name: PyTest with linters
+        command: venv/bin/pytest
+    - run:
+        name: Lint YAML
+        command: venv/bin/yamllint -c .yamllint.yaml .
+    - save_cache:
+        paths:
+        - venv/
+        key: *cache_key
+  verify-requirements:
+    docker: *docker
+    steps:
+    - checkout
+    - run:
+        name: Verify that requirements.txt contains the right dependencies for
+          this python version
+        # use `--constraint` with `requirements.in` not `requirements.txt`
+        # because for pip>=20.3 "Constraints are only allowed to take the form
```

Comment:
Just to be clear, this is because we have hashes in `requirements.txt`?

Reply:
yes